docs/docs/forms.mdx

```diff
@@ -361,75 +361,151 @@ to fully customize the validation process.
 If none of the predefined [Slot Mappings](./forms.mdx#slot-mappings) fit your use
 case, you can use the
 [Custom Action](./actions.mdx#custom-actions) `validate_{form_name}` to write your own
-extraction code. Rasa Open Source will trigger this function when the form is run.
+extraction code. Rasa Open Source will trigger this action when the form is run.
+
+If you're using the Rasa SDK we recommend you to extend the provided
+`FormValidationAction`. When using the `FormValidationAction`, two things are needed to
+extract customs slots:
+
+1. Override `required_slots`. It should return a list of slot names which your form
+   requires to be filled.
+2. Define a method `extract_<slot_name>` for every custom slot which you want to
+   extract.
 
```

Comment:
I'm not sure that these two are dependent. Isn't it possible to dynamically generate the required_slot list without needing to define any `extract_<slot_name>` methods or has that changed in Rasa 2.0? It may help to make a clearer distinction between these two use cases, so users don't think they need to define these for every slot.

Reply:
> without needing to define any extract_<slot_name> methods 

@kearnsw Not sure I'm understanding this part. You'll always need to define an `extract_<slot name>` method for custom extractions. We could indeed dynamically generate `required_slots` but I'd rather go for explicitness than magic here. @samsucik and I iterated a bit more on the solution and updated the PR accordingly. Hope it makes more sense now.